drivers/sensor/bme280/bme280.c

```diff
@@ -318,6 +322,12 @@ static int bme280_sample_fetch(const struct device *dev,
 
 	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	/* Do not allow sample fetching from OFF state */
+	if (data->pm_state == DEVICE_PM_OFF_STATE)
+		return -EIO;
```

Comment:
Is it really an error? Or just a simple return with maybe LOG_WRN?

Reply:
Well, in the OFF state the device is inactive so any attempt to fetch data from it should fail. I think that the app should get a notification of that. I can change the return code if you find a different one more appropriate.

Reply:
On a second thought I think you are right. You can leave -EIO